routers/api/v1/repo/pull.go

```diff
@@ -966,8 +966,33 @@ func parseCompareInfo(ctx *context.APIContext, form api.CreatePullRequestOption)
 		return nil, nil, nil, nil, "", ""
 	}
 
+	var rootRepo *repo_model.Repository
+	if baseRepo.IsFork && !isSameRepo {
+		err = baseRepo.GetBaseRepo()
+		if err != nil {
+			if !repo_model.IsErrRepoNotExist(err) {
+				ctx.ServerError("Unable to find root repo", err)
+				return nil, nil, nil, nil, "", ""
+			}
+		} else {
+			rootRepo = baseRepo.BaseRepo
```

Comment:
Should we still find `BaseRepo`'s base repository?

Reply:
Hmm, that's so old that I has forgoten why ...,  maybe it's just be copyed from here ...
https://github.com/go-gitea/gitea/blob/e81ccc406bf723a5a58d685e7782f281736affd4/routers/web/repo/compare.go#L340-L351